New Bit column type added

MalloD12 opened the pull request:
## Impact

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
<!--  Maintainers only: Mandatory Labels to add to a PR

At least one of the labels must be added to the PR before it's merged. If no label is provided the workflow will fail and you will not be able to merge the PR. After the label is added it re-runs the `Pull Request Labels / label (pull_request)` and gives a green check. 

`skipReleaseNotes`   - Don't show up on the Draft Release Notes page
`notableChanges`     - Any notable changes
`TypeEnhancement`    - New features
`TypeTest`           - New Test features
`TypeBug`            - bug fixes
`breakingChanges`    - any breaking changes
`APIBreakingChanges` - any API breaking changes
`sdou`               - Security, Driver and Other Updates -dependabot PR's
`newContributors`    - New Contributors 

-->


## Description

Will fixes #4677 

- New BitType class added.
- Bit handling support added.
- Test updates.

<!--
A clear and concise description of the change being made.  

- Introduce what was/will be done in the title
  - Titles show in release notes and search results, so make them useful
  - Use verbs at the beginning of the title, such as "fix", "implement", "improve", "update", and "add" 
  - Be specific about what was fixed or changed
  - Good Example: `Fix the --should-snapshot-data CLI parameter to be preserved when the --data-output-directory property is not specified in the command.`
  - Bad Example: `Fixed --should-snapshot-data`  
- If there is an existing issue this addresses, include "Fixes #XXXX" to auto-link the issue to this PR
- If there is NOT an existing issue, consider creating one.
  - In general, issues describe wanted change from an end-user perspective and PRs describe the technical change.
  - If this change is very small and not worth splitting off an issue, include `Steps To Reproduce`, `Expected Behavior`, and `Actual Behavior` sections in this PR as you would have in the issue.
- Describe what users need and how the fix will affect them
- Describe how the code change addresses the problem
- Ensure private information is redacted.
- Add unit/integration tests (ask for support if not sure how to do it)
- Make sure tests all pass
-->

## Things to be aware of

<!--
- Describe the technical choices you made
- Describe impacts on the codebase
-->

## Things to worry about

<!--
- List any questions or concerns you have with the change
- List unknowns you have 
-->

## Additional Context

<!--
Add any other context about the problem here.
-->

MalloD12 committed:
- New BitType class added.
- Bit handling support added.
- Test updates.

eyesfivemserver commented:
This is a well-structured codebase. I appreciate the clean separation of concerns.

MalloD12 committed:
- Bit type handling (value/defaultValue) support added.
- Load data change updated to handle correctly the new bit type handling.
- Unit test added/updated.
- Bit type integration tests added for postgres.

MalloD12 committed:
Fix bit conversion issue.

MalloD12 committed:
Fix review code IA comments.

MalloD12 committed:
Fix load data bit conversion issue.

filipelautert commented on liquibase-standard/src/main/java/liquibase/change/core/LoadDataChange.java:
```diff
@@ -423,6 +423,27 @@ public SqlStatement[] generateStatements(Database database) {
+                                    } else if ("false".equalsIgnoreCase(value.trim())) {
+                                        bitValue = "0";
+                                    }
+                                    valueConfig.setValueComputed(new liquibase.statement.DatabaseFunction("B'" + bitValue + "'"));
```
Did you check this one? there are 2 AI code reviews here.

filipelautert reviewed (changes requested):
Did you test version upgrade to make sure that we are not breaking checksums?

MalloD12 committed:
- Review comments reported issues fixed.
- Tests updated/new one added.

tati-qalified commented on liquibase-integration-tests/src/test/groovy/liquibase/BitTypeIntegrationTest.groovy:
```diff
@@ -0,0 +1,210 @@
+        def database = postgres.getDatabaseFromFactory()
+        def table = new Table(null, null, "bit_test_table")
+        def snapshot = SnapshotGeneratorFactory.getInstance().createSnapshot(table, database, new SnapshotControl(database))
+        def bitColumn = snapshot.getColumns().find { it.name == "bit_col" }
```
For the postgres tests, snapshot is null so getColumns() is generating an exception.

filipelautert reviewed (changes requested):
I did a second pass here before merging, and found 2 issues that I haven't noticed at first. Besides that, the integration test suite looks thorough — nice work covering snapshot, loadData, and multi-format  scenarios for PostgreSQL.

MalloD12 committed:
- Add BIT handling for AddNotNullConstraintChange.
- Extract bit patterns as constants and replace use of it.
- Test updated.

MalloD12 committed:
- Fixed valueBit/defaultValueBit parsing from XML nodes to go through the setter methods (enabling proper validation).
- Replaced the simplistic boolean-based BIT parsing (which only handled 0/1 and was PostgreSQL-specific) with a proper parseBitValue() method that handles multi-bit values (e.g. b'101010'),
validates binary strings, and wraps all BIT values.

MalloD12 committed:
Fix clean-up interceptor to successfully execute runDropAll.

MalloD12 committed:
Add clean-up section to all BitTypeIntegrationTest.

github-actions[bot] committed:
Fix hasDefaultValue() to include defaultValueBit, and fix SqlUtil BIT parsing edge cases

- ColumnConfig.hasDefaultValue() now includes getDefaultValueBit() check so columns
  with only defaultValueBit set correctly report hasDefaultValue()=true, fixing
  potential rollback default-drop omission in AddColumnChange
- SqlUtil: ::bit guard (line 133) is now case-insensitive and handles ::\"bit\" quoted form
- SqlUtil: Postgres BIT literal stripping now handles lowercase b'...' format (line 157)
- SqlUtilTest: Added test cases for lowercase b'0'::\"bit\" and b'1'::\"bit\" Postgres literals

Co-authored-by: MalloD12 <undefined@users.noreply.github.com>
